File: apps/Lodgr.Api/Models/DepositTransaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Lodgr.Api.Models;

[Table("deposit_transactions")]
public class DepositTransaction
{
    [Key]
    [Column("deposit_txn_id")]
    public long DepositTxnId { get; set; }

    [Column("contract_id")]
    public long ContractId { get; set; }

    [Column("txn_type")]
    public TxnType TxnType { get; set; }

    [Range(typeof(decimal), "0.01", "79228162514264337593543950335")]
    [Column("amount", TypeName = "numeric(12,2)")]
    public decimal Amount { get; set; }

    [Column("transaction_date")]
    public DateOnly TransactionDate { get; set; }

    [Column("payment_method")]
    public PaymentMethod? PaymentMethod { get; set; }

    [StringLength(100)]
    [Column("reference_code")]
    public string? ReferenceCode { get; set; }

    [Column("recorded_by")]
    public long? RecordedBy { get; set; }

    [Column("note")]
    public string? Note { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}

public enum TxnType
{
    COLLECT,
    REFUND,
    DEDUCTION,
    ADJUSTMENT,
}

File: apps/api/src/main/java/io/github/hoanghonghuy/lodgr/entity/DepositTransaction.java
package io.github.hoanghonghuy.lodgr.entity;

import io.github.hoanghonghuy.lodgr.enums.DepositTransactionType;
import io.github.hoanghonghuy.lodgr.enums.PaymentMethod;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "deposit_transactions")
@Getter
@Setter
public class DepositTransaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "deposit_txn_id")
    private Long depositTransactionId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "contract_id", nullable = false)
    private Contract contract;

    @Enumerated(EnumType.STRING)
    @Column(name = "txn_type", nullable = false)
    private DepositTransactionType transactionType;

    @Column(name = "amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(name = "transaction_date", nullable = false)
    private LocalDate transactionDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_method")
    private PaymentMethod paymentMethod;

    @Column(name = "reference_code")
    private String referenceCode;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recorded_by")
    private User recordedBy;

    @Column(name = "note")
    private String note;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
}
